src/responder/kcm/secrets/secrets.c

```diff
@@ -181,11 +188,166 @@ static struct ldb_dn *per_uid_container(TALLOC_CTX *mem_ctx,
     return uid_base_dn;
 }
 
+static errno_t get_secret_expiration_time(uint8_t *key, size_t key_length,
+                                          uint8_t *sec, size_t sec_length,
+                                          time_t *_expiration)
+{
+    errno_t ret;
+    TALLOC_CTX *tmp_ctx;
+    time_t expiration = 0;
+    struct cli_creds client = {};
+    struct kcm_ccache *cc;
+    struct sss_iobuf *iobuf;
+    krb5_creds **cred_list, **cred;
+    const char *key_str;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        return ENOMEM;
+    }
+
+    key_str = talloc_strndup(tmp_ctx, (const char *) key, key_length);
+    if (key_str == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    iobuf = sss_iobuf_init_readonly(tmp_ctx, sec, sec_length);
+    if (iobuf == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    ret = sec_kv_to_ccache_binary(tmp_ctx, key_str, iobuf, &client, &cc);
+    if (ret != EOK) {
+        goto done;
+    }
+
+    cred_list = kcm_cc_unmarshal(tmp_ctx, NULL, cc);
+    if (cred_list == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    for (cred = cred_list; *cred != NULL; cred++) {
+        if ((*cred)->times.endtime != 0) {
+            expiration = (time_t) (*cred)->times.endtime;
+        }
+    }
+
+    if (_expiration != NULL) {
```

Comment:
Calling this function without `_expiration` parameter does not make much sense. I would check that at the beginning and return EINVAL if it is NULL. wdyt?

Reply:
You are right